src/bun.js/webcore/streams.zig

```diff
@@ -4541,7 +4543,7 @@ pub const File = struct {
         var fd = if (file.pathlike != .path)
             // We will always need to close the file descriptor.
             switch (Syscall.dup(file.pathlike.fd)) {
-                .result => |_fd| _fd,
+                .result => |_fd| if (Environment.isWindows) bun.toLibUVOwnedFD(_fd) else _fd,
```

Comment:
Learning bit for me: When does something need to be `bun.toLibUVOwnedFD(_fd)` vs. say a regular `bun.toFD` bun descriptor?

Reply:
from my understanding native windows fds can't be passed to libuv functions. in this case the downstream code uses libuv to do the reading, so we have to convert the fd before using it

Reply:
FileDescriptor on windows is a union type between a UV/C-runtime fd int (c_int) and a Windows handle (*anyopaque)

converting uv -> windows is ok but uv owns the descriptor
converting windows -> uv moves ownership to uv/crt.

bun.toLibUVOwnedFD consumes the input FileDescriptor and returns a new FileDescriptor that is always a libuv one. it is expected that this new one will eventually be `close()`d